plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -17,149 +25,52 @@
-             compared to unordered categorical values
-         - ":"-separated values for ranges of scalars.
-           "int" and/or "log" modifiers can be added in front to cast to integer or
-           use log-distributed values (Eg: int:log:4:1024)
-         - anything else will be treated as a constant string
-       * a config definition dict for scalar parameters, with potential fields
-         init, lower, upper, step, log, integer
-       * a list for option parameters defined in config file
-
-    Returns
-    -------
-    Parameter or str
-        A Parameter if the string fitted one of the definitions, else the input string
-    """
-    # lazy initialization to avoid overhead when loading hydra
-    import nevergrad as ng
-
-    # revert config parsing
-
-    if isinstance(description, (ListConfig, list)):
-        description = ",".join(str(x) for x in description)
-    if isinstance(description, str):
-        # cast to spec if possible
-        try:
-            description = CommandlineSpec.parse(description)
-        except ValueError:
-            pass
-    # convert scalar commandline specs to dict
-    if isinstance(description, CommandlineSpec) and description.bounds is not None:
-        description = ScalarConfigSpec(
-            lower=description.bounds[0],
-            upper=description.bounds[1],
-            log=description.log,
-            integer=description.cast == "int",
-        )
-    # convert scalar config specs to dict
-    # convert dict to Scalar parameter instance
+def create_nevergrad_parameter(description: Any) -> Any:
+    scalar = None
+    if isinstance(description, Override):
+        override = description
+        val = override.value()
+        if override.is_sweep_override():
+            if override.is_choice_sweep():
+                val = cast(ChoiceSweep, val)
+                vals = [
+                    x for x in override.sweep_iterator(transformer=Transformer.encode)
+                ]
+                if "ordered" in val.tags:
+                    return ng.p.TransitionChoice(vals)
+                else:
+                    return ng.p.Choice(vals)
+            elif override.is_range_sweep():
+                # if shuffled or integer range and <=6, use Choice
+                val = cast(RangeSweep, val)
+                if val.shuffle or (
+                    val.stop is not None
+                    and val.start is not None
+                    and val.stop - val.start <= 6
+                ):
+                    vals = [
+                        x
+                        for x in override.sweep_iterator(transformer=Transformer.encode)
+                    ]
+                    return ng.p.Choice(vals)
+                params = {"lower": val.start, "upper": val.stop}
+                scalar = ng.p.Scalar(**params)  # type: ignore
+                if isinstance(val.start, int):
+                    scalar.set_integer_casting()
```

Comment:
Initially I was trying to stay more consistent with the original impl... but i think your suggestion is better, it will also be easier for users to use: more straightforward and no surprieses.